work/sixth/index.html

```diff
@@ -16,6 +16,72 @@
   <script src="../../common/js/jquery.js"></script>
   <script>
     // 回答はここに書いてね
+    let encountHtml =
+      `
+        <p class="name"></p>
+        <div>
+          <img class="img" src="" alt="">
+        </div>
+        <div>
+          <button class="btn" id="catch">捕まえる</button>
+          <button class="btn" data-js="reset">逃げる</button>
+        </div>
+      `
+      
+    let failHtml =
+      `
+        <p class="text">逃げられてしまった...</p>
+        <button class="btn mt" data-js="reset">戻る</button>
+      `
+
+    let doneHtml =
+      `
+        <p class="name">ゲットしました</p>
+        <div>
+          <img class="img" src="" alt="">
+        </div>
+        <button class="btn" data-js="reset">戻る</button>
+      `
+
+    $(function() {
+      let defaultHtml = $('.wrapper').clone(true);
+      function resetHtml() {
+        $(document).on('click', '.btn[data-js]', function() {
+          $('.wrapper').replaceWith(defaultHtml.clone());
+        })
+      };
+      $(document).on('click', '#catch', function() {
+        $.ajax({
+          url: 'https://pokeapi.co/api/v2/item/4',
+          method: 'GET',
+        }).done(function(res) {
+          var ballImg = res.sprites.default;
+          var selectTemplate = (Math.random() >0.5) ? doneHtml : failHtml;
+          $('.wrapper').html(selectTemplate);
+          $('.img').attr('src', ballImg);
+          resetHtml();
+        });
+      });
```

Comment:
上記コメントと同様、他のクリックイベントに渡すコールバック関数も関数化してあげましょう！
そうすると、下記のサジェストのように長い処理が1行にできます。
```suggestion
      $(document).on('click', '#catch', ⭐︎第三引数に渡すコールバック関数を関数化する);
```

Reply:
こちらも上記と同様のコミットで、処理内容を関数化したしたものをクリックイベント時に実行するよう書き換えました。
7678d68686b9abdcdf166891420bd5fe70975a09